marklogic-data-hub/src/main/java/com/marklogic/hub/step/StepDefinition.java

```diff
@@ -62,8 +62,25 @@ public String toString() {
      * @param type - the type of the mapping
      * @return a Step object
      */
-    static Step create(String name, StepType type) {
-        return new StepImpl(name, type);
+    static StepDefinition create(String name, StepType type) {
+        StepDefinition stepDefinition = null;
+
+        switch (type) {
+            case INGEST:
+                stepDefinition = new IngestionStepDefinitionImpl(name);
```

Comment:
Didn't this change to "INGESTION" and "MASTERING" ?

Reply:
Didn't want to break the current code.
Should be taken up as part of https://project.marklogic.com/jira/browse/DHFPROD-2096